### PR TITLE
build: drop minimum supported node version to 16.13.0

### PR DESCRIPTION
**Pull request by RebeccaStevens**
## PR Checklist

- [ ] Addresses an existing open issue
- [ ] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/ts-api-utils/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/ts-api-utils/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

Node 16 is still in lts so we should probably still support it.
Node `16.13.0` is the oldest version of node in lts.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -62,7 +62,8 @@ jobs:
                 required_pull_request_reviews: null,
                 required_status_checks: {
                   checks: [
-                    { context: "build" },
+                    { context: "build (16.13.0)" },
+                    { context: "build (18.x)" },
```
Comment (RebeccaStevens) [inline .github/workflows/release.yml:66]: I think these is the right ids.
Having a space in there though makes me doubt though.

Comment (JoshuaKGoldberg) [reply .github/workflows/release.yml:66]: Heh, yeah, I don't know. Guess we'll find out the hard way if we need! 😄

**Review by JoshuaKGoldberg — APPROVED** [review]
Agreed, thanks!

**Comment by JoshuaKGoldberg** [comment]
@allcontributors add @RebeccaStevens for project management

**Comment by JoshuaKGoldberg** [comment]
@allcontributors add @JoshuaKGoldberg for project management

**Comment by JoshuaKGoldberg** [comment]
@allcontributors add @RebeccaStevens for docs

**Comment by JoshuaKGoldberg** [comment]
@allcontributors add @RebeccaStevens for tests

**Comment by JoshuaKGoldberg** [comment]
@allcontributors add @RebeccaStevens for tool

**Pull request merged**